contents/textbook/lecture05/boundedWildcards/bounedWildcards.md

```diff
@@ -0,0 +1,67 @@
+<frontmatter>
+  pageNav: 2
+  header: header.md
+  footer: footer.md
+  siteNav: site-nav.md
+</frontmatter>
+
+<br> 
+
+# bounded Wildcards
+<br> 
+
+<!-- DO NOT DELETE THIS LINK AND PLEASE WRITE BELOW THIS LINK-->
+[Edit the material here! :fas-pen:](https://github.com/nus-cs2030/1920-s2/edit/master/contents/textbook/lecture05/unboundWildcards/unboundWildcards.md)
+<!-- DO NOT DELETE THIS LINK AND PLEASE WRITE BELOW THIS LINK-->
+
+This was adapted from [The Java tutorial](https://docs.oracle.com/javase/tutorial/java/generics/upperBounded.html) and [Question 445](https://piazza.com/class/k54zo22zq1t2zc?cid=445) from Piazza.
+
+> **upper-bounded wildcard**
+
+`<? extends T>` is known as the upper-bounded wildcard. This is used to relax the use of complex type to its **co-variant complex types**, i.e.complex types with type parameter being subtype of `T`.  So, `List<? extends T>` means **a List of subtypes of T**. 
+Short example:
+consider the following `process` method:
+```
+public static void process(List<? extends T> myList) {
+    for (T ele : myList) {
+        // ...
+    }
+}
+```
+`myList` is the a list of **T** or **subtype of T**, thus no matter what runtime-type `ele` is, we can always assign compile time type **T** to it.
+
+> **lower-bounded wildcard**
+
+Similarly, `<? extends T>` is known as the lowwer-bounded wildcard, which relaxs the use of complex type to its **contra-variant complex types**, i.e.complex types with type parameter being supertype of `T`.  So, `List<? extends T>` means **a List of supertypes of T**.
```

Comment:
You mean <? super T> here, don't you?

Reply:
Yea @NoooooobAlex Do fix the typo! 👍 

Once done, we will merge in your pull request thank you!

Reply:
Ok. (Sorry for late response....)

Reply:
It's alright! Thank you XD 💯